### PR TITLE
[Prot Paladin] Add First Avenger and Moment of Glory analyzers, minor cleanup

### PR DESCRIPTION
**Pull request by hpabst**
# Notes
- Fix some broken spell IDs for protection paladin.
- Remove some leftover console.log calls from a previous PR. Whoops.
- Add analyzers to track bonus damage dealt by taking First Avenger and Moment of Glory talents.
- [Log used for First Avenger](https://www.warcraftlogs.com/reports/KzFbPaBqAgr9h4Rw#fight=11&type=summary&source=8)
- [Log used for Moment of Glory](https://www.warcraftlogs.com/reports/N4gKnBz6XpGdZjax#fight=7&type=summary&source=87)

# Screenshots
2020-10-26 11_28_26-Mythic The Hivemind - Kill (4_13) by Ashincka in Battle of Dazar'alor - WoWAnaly
2020-10-26 11_28_50-Mythic The Hivemind - Kill (4_13) by Ashincka in Battle of Dazar'alor - WoWAnaly
2020-10-26 11_29_21-Mythic The Hivemind - Kill (3_51) by Fatpala in Mythic - WoWAnalyzer
2020-10-26 11_29_31-Mythic The Hivemind - Kill (3_51) by Fatpala in Mythic - WoWAnalyzer

### CODE REVIEW

File: src/parser/paladin/protection/modules/talents/FirstAvenger.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    this.totalNumCasts += 1;
+  }
+
+  trackAvengersShieldHits(event: DamageEvent): void {
+    if (event.timestamp - this.lastAvengersShieldCastTimestamp < AVENGERS_SHIELD_BOUNCE_BUFFER) {
+      if (!this.castToHitsMap.has(this.lastAvengersShieldCastTimestamp)) {
+        this.castToHitsMap.set(this.lastAvengersShieldCastTimestamp, []);
+      }
+      this.totalNumHits += 1;
+      const hits: DamageEvent[] | undefined = this.castToHitsMap.get(this.lastAvengersShieldCastTimestamp);
```
Comment (joshinat0r) [inline src/parser/paladin/protection/modules/talents/FirstAvenger.tsx:43]: This (and pushing into `hits`) *shouldn't* do anything, some kind of leftover from a previous version?

File: src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        position={STATISTIC_ORDER.DEFAULT}
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={(
+          <>
+            You hit <b>{formatNumber(this.damageBoostedHits)}</b> targets with a <SpellLink id={SPELLS.AVENGERS_SHIELD.id} /> boosted with <SpellLink id={SPELLS.MOMENT_OF_GLORY_TALENT.id} />.
+          </>
+        )}
+      >
+       <BoringSpellValue
```
Comment (joshinat0r) [inline src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx]: Needs some space
image

**Review by joshinat0r — CHANGES_REQUESTED** [review]

File: src/parser/paladin/protection/modules/talents/FirstAvenger.tsx (hunk 1 above)
Comment (hpabst) [reply src/parser/paladin/protection/modules/talents/FirstAvenger.tsx:43]: I'm a bit shaky on how Javascript handles array references (I'm a backend Java dev myself), but this roundabout way of pushing a `DamageEvent` into the map value is succeeding according to the debug logs I put in place (adding a `console.dir` to the `statistic()` call). The `Map` values are then used in calculating the extra damage from `Avenger's Shield` that can be attributed to `First Avenger`.

image

File: src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx (hunk 2 above)
Comment (hpabst) [reply src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx]: Can you elaborate? I'm not sure what you mean.

File: src/parser/paladin/protection/modules/talents/FirstAvenger.tsx (hunk 1 above)
Comment (joshinat0r) [reply src/parser/paladin/protection/modules/talents/FirstAvenger.tsx:43]: You're right, my bad. Not a primitive, thats why its passed by reference.
Said nothing

File: src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx (hunk 2 above)
Comment (joshinat0r) [reply src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx]: The red lines are missing a space. Indentation is 2 but the amount of spaces in those lines are uneven

Comment (hpabst) [reply src/parser/paladin/protection/modules/talents/MomentOfGlory.tsx]: Resolved in the latest commit.

**Review by joshinat0r — APPROVED** [review]

**Pull request merged**